### DISCUSSION TITLE
Encrypted videos download as corrupted/non-functional (Firefox, Windows)

### DISCUSSION THREAD

**Discussion by hobart3052** [aclap-dev/video-downloadhelper#1309] [Help and Q&A]
### Your question

In the past using this extension (same browser & computer) I was able to download anything, but now videos on sites like OF will still "download", but the resulting file is a broken video that just shows flashing colors. Is there any way to fix this? Please don't comment if you're just going to talk about legality. 

### Addon & CoApp version and details

version: 9.1.0.44
target: mozilla
channel: stable
lang: en-US
coapp: {"found":true,"path":"C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:130.0) Gecko/20100101 Firefox/130.0
{
    "download_directory": "C:\\Users\\waltb\\dwhelper",
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 10,
    "successfull_dl": 26,
    "download_history": {
        "__serializer_tag": "map",
        "__serializer_value": []
    }
}

**Comment by Mattmjs** [comment]
No way to fix it, it's DRM which scrambles the file.
Like it or not that's how DRM works, the people on
only fans don't want you copying their content even if you paid to access it. I'm not aware of other software circumventing it. If you want it,try 
screen recording it.

**Reply by GrampaWildWilly** [reply]
I do not work for VDH.  I'm just another user, like Matt.

This has been reported a number of times before.  You need to understand that DRM exists for the express purpose of preventing the downloading of copyrighted material.  VDH has never, does not now, & will never break DRM protection.  That would be illegal.  Since you are using Firefox, you will see a small icon at the top of the browser window in the same line as the URL, to the left of the https://.  This icon looks like a chain link.  Hover your mouse over it to get it to hover up a tooltip.  The icon indicates that the content on that web page is protected by DRM.  Look for that icon on every web page.  Some pages will have it, some will not.  Some pages on OnlyFans will have it, some will not.  It seems to be at the whim of the author whether the content is protected by DRM.  Refer to #1306 for another similar example.

On the subject of screen recording, a few months ago I went through a futile exercise looking for a screen recorder that would record content protected by DRM.  I couldn't find one among the approximately 2 dozen I tried.  Every single one of them either caused the video that should have been recorded to go completely black during playback, or went ahead & recorded something that turned out to be just a black screen.  It's a conspiracy.  I think it's illegal.  There was a class action lawsuit about 50 or so years ago that decided that people were legally permitted to record whatever appeared on their televisions.  This legal decision gave rise to the videocassette recorder industry.  I believe we are looking at the same situation today.  It's going to take another lawsuit to get rid of DRM.  I advise against holding your breath for this lawsuit to occur.

**Reply by masterofobzene** [reply]
whatever you say, but aria2c still downloads the SAME url from the video (copied FROM video downloadhelper extension) without any corruption.

I ended up here because I saw the corruption happening, then immediately tried downloading the same video, from the same page using the same link copied from this extension, using aria2c and the video downloaded perfectly. 

When corruption happens, I simply do the same: copy the download link without triggering the download using this extension, then paste the link into aria2c. Inconvenient but working.
